### PR TITLE
[release-17.0] Use Debian Bullseye in Bootstrap

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR is a follow-up of https://github.com/vitessio/vitess/pull/13664. We are bumping the `bootstrap` images to `bullseye` in order to keep up with the Golang upgrades. We are not changing the packages we install, we keep the `buster` packages. Except on `percona` images, where `bullseye` packages had to be installed for compatibility.

I have tested the entire operator example back and forth with different MySQL version and was able to complete it each time.

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: docker/bootstrap/Dockerfile.percona57 [hunk 1]
```diff
@@ -5,16 +5,15 @@ FROM --platform=linux/amd64 "${image}"
...
         echo debconf debconf/frontend select Noninteractive; \
         echo percona-server-server-5.7 percona-server-server/root_password password 'unused'; \
         echo percona-server-server-5.7 percona-server-server/root_password_again password 'unused'; \
     } | debconf-set-selections && \
     apt-get update && \
-    apt-get install -y --no-install-recommends \
-        percona-server-server-5.7 \
-	libperconaserverclient20-dev percona-xtrabackup-24 && \
+    apt-get install -y --no-install-recommends percona-server-server-5.7 && \
+    apt-get install -y --no-install-recommends libperconaserverclient20-dev percona-xtrabackup-24 && \
```
Comment (GuptaManan100) [inline docker/bootstrap/Dockerfile.percona57:16]: Could we not merge these commands?

**Review by GuptaManan100 — COMMENTED** [review]
Why are we changing from `buster` to `bullseye`

**Comment by frouioui** [comment]
> Changing from `buster` to `bullseye` doesn't affect the functionality of the docker images, right?

After running some tests it did not.

File: docker/bootstrap/Dockerfile.percona57 (hunk 1 above)
Comment (frouioui) [reply docker/bootstrap/Dockerfile.percona57:16]: That led to unexpected behaviors where `percona-xtrabackup-24` was going through an apt-get setup phase was left pending for hours. Splitting the command in two solved the issue.

**Review by GuptaManan100 — APPROVED** [review]
Approving since manual testing has been done 👍

**Review by vmg — APPROVED** [review]

**Pull request merged**